src/ansys/fluent/core/launcher/container_launcher.py

```diff
@@ -153,24 +153,26 @@ def __init__(
         del argvals["self"]
         if argvals["start_timeout"] is None:
             argvals["start_timeout"] = 60
-        for arg_name, arg_values in argvals.items():
-            setattr(self, arg_name, arg_values)
         self.argvals = argvals
-        self.new_session = self.mode.value[0]
+        self.new_session = self.argvals["mode"].value[0]
         self.file_transfer_service = file_transfer_service
 
     def __call__(self):
-        if self.mode == FluentMode.SOLVER_ICING:
+        if self.argvals["mode"] == FluentMode.SOLVER_ICING:
             self.argvals["fluent_icing"] = True
         args = _build_fluent_launch_args_string(**self.argvals).split()
-        if FluentMode.is_meshing(self.mode):
+        if FluentMode.is_meshing(self.argvals["mode"]):
             args.append(" -meshing")
-        if self.container_dict is None:
-            setattr(self, "container_dict", {})
-        if self.product_version:
-            self.container_dict["image_tag"] = f"v{self.product_version.value}"
-        if self.dry_run:
-            config_dict, *_ = configure_container_dict(args, **self.container_dict)
+        if self.argvals["container_dict"] is None:
+            self.argvals["container_dict"] = {}
+        if self.argvals["product_version"]:
+            self.argvals["container_dict"][
+                "image_tag"
+            ] = f"v{self.argvals['product_version'].value}"
+        if self.argvals["dry_run"]:
+            config_dict, *_ = configure_container_dict(
+                args, **self.argvals["container_dict"]
+            )
```

Comment:
Just wondering why some arguments are massaged in `__init__()`, others in `__call__`. The default would normally be `__init__()`.

I'll explain why this will become particularly pertinent here. The idea here is that users will use 
 and reuse _launcher_  _command objects_ directly (without going via `launch_fluent`). Before executing a command, the user should be able to query its argument state and see the argument values that will actually be used. 

@mkundu1 @prmukherj @hpohekar @raph-luc

Reply:
Yes, the value of `self.argvals` should be final (as close as possible) at the end of `__init__()`.

Reply:
Yes, I agree. Therefore I already explained in the previous comments ([on closed PR](https://github.com/ansys/pyfluent/pull/2820)) that we are using only `self.<argument_name>` inside `__call__` method not `self.argvals["<argument_name>"]` simultaneously.

![image](https://github.com/ansys/pyfluent/assets/106588300/2b91302c-0489-477c-bb40-337d2579cf49)

Reply:
I believe that is not the point being raised here @hpohekar. What I believe they are saying is that all this handling of `self.argvals` being done in `__call__()` should instead be done in `__init__()`, and I think that is a very good point.

Another benefit is that moving this logic to `__init__()` will save on having to prefix `self.` before every `argvals` reference, as `argvals` is a local variable there. 

So, for example, rather than having this in `__call__()`:
```
if self.argvals["mode"] == FluentMode.SOLVER_ICING:
    self.argvals["fluent_icing"] = True
```
We would have this in `__init__()`:
```
if argvals["mode"] == FluentMode.SOLVER_ICING:
    argvals["fluent_icing"] = True
```
(before `self.argvals = argvals`)

Reply:
Done. Thank you.